File: app/src/main/java/com/corporation8793/medicinal_herb/activity/join/JoinActivity.java
package com.corporation8793.medicinal_herb.activity.join;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.InputFilter;
import android.text.Spanned;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Toast;

import com.corporation8793.medicinal_herb.R;
import com.corporation8793.medicinal_herb.activity.LoginActivity;

import java.util.regex.Pattern;

public class JoinActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_join);

        EditText id_input_box  = findViewById(R.id.id_input_box);
        EditText pw_input_box  = findViewById(R.id.pw_input_box);
        EditText pw_input_check_box  = findViewById(R.id.pw_check_input_box);
        CheckBox access_term_check = findViewById(R.id.access_term_check);

        LinearLayout next_btn = findViewById(R.id.next_btn);
        next_btn.setOnClickListener(v->{
            if(id_input_box.getText().toString().trim().equals("")){
                Toast.makeText(getApplicationContext(),"아이디를 입력해주세요.", Toast.LENGTH_SHORT).show();
            } else if (!pw_input_box.getText().toString().equals(pw_input_check_box.getText().toString())){
                Toast.makeText(getApplicationContext(),"비밀번호가 일치하는지 다시 한번 확인해주세요.", Toast.LENGTH_SHORT).show();
            } else if (!access_term_check.isChecked()){
                Toast.makeText(getApplicationContext(),"이용약관을 동의해주세요.", Toast.LENGTH_SHORT).show();
            }else{
                Intent intent = new Intent(JoinActivity.this, ProfileActivity.class);
                startActivity(intent);
            }

        });

        InputFilter filterAlphaNum = new InputFilter() {
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
                Pattern ps = Pattern.compile("^[a-zA-Z0-9]+$");
                if (!ps.matcher(source).matches()) {
                    Toast.makeText(getApplicationContext(),"영어와 숫자만 적어주세요.",Toast.LENGTH_SHORT).show();
                    return id_input_box.getText().toString()+"";
                }
                return null;
            }
        };

// 정규표현식만 적용하는 경우
        id_input_box.setFilters(new InputFilter[] {
                filterAlphaNum
        });
    }
}

File: app/src/main/java/com/corporation8793/medicinal_herb/activity/join/JoinActivity.kt
package com.corporation8793.medicinal_herb.activity.join


import android.app.Dialog
import android.content.Intent
import android.graphics.Color
import android.graphics.Point
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.text.InputFilter
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintLayout
import com.corporation8793.medicinal_herb.Common
import com.corporation8793.medicinal_herb.R
import java.util.regex.Pattern

class JoinActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_join)
        val id_input_box = findViewById<EditText>(R.id.id_input_box)
        val pw_input_box = findViewById<EditText>(R.id.pw_input_box)
        val pw_input_check_box = findViewById<EditText>(R.id.pw_check_input_box)
        val access_term_check = findViewById<CheckBox>(R.id.access_term_check)
        val next_btn = findViewById<LinearLayout>(R.id.next_btn)
        val access_terms_box = findViewById<ConstraintLayout>(R.id.access_terms_box)
        next_btn.setOnClickListener { v: View? ->
            if (id_input_box.text.toString().trim { it <= ' ' } == "") {
                Toast.makeText(applicationContext, "아이디를 입력해주세요.", Toast.LENGTH_SHORT).show()
            } else if (pw_input_box.text.toString() != pw_input_check_box.text.toString()) {
                Toast.makeText(applicationContext, "비밀번호가 일치하는지 다시 한번 확인해주세요.", Toast.LENGTH_SHORT).show()
            } else if (!access_term_check.isChecked) {
                Toast.makeText(applicationContext, "이용약관을 동의해주세요.", Toast.LENGTH_SHORT).show()
            } else {
                val intent = Intent(this@JoinActivity, ProfileActivity::class.java)
                intent.putExtra("type","join")
                intent.putExtra("id",id_input_box.text.toString())
                intent.putExtra("pw",pw_input_box.text.toString())
                startActivity(intent)
            }
        }
        val filterAlphaNum = InputFilter { source, start, end, dest, dstart, dend ->
            val ps = Pattern.compile("^[a-zA-Z0-9]+$")
            if (!ps.matcher(source).matches()) {
                Toast.makeText(applicationContext, "영어와 숫자만 적어주세요.", Toast.LENGTH_SHORT).show()
            }
            null
        }

        access_terms_box.setOnClickListener{
            Common().showAccessTerms(this)
        }

// 정규표현식만 적용하는 경우
        id_input_box.filters = arrayOf(
                filterAlphaNum
        )




    }







//        dialog.getWindow()!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT));

//        dialog.getWindow()!!.clearFlags(WindowManager.LayoutParams.FLAG_DIM_BEHIND);



}